### PR TITLE
BH-143: Upgrade to composer v2

### PR DESCRIPTION
**Pull request by mmetayer**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Review by jmleroux — APPROVED** [review]

File: .circleci/detect_structure_changes.sh [hunk 1]
```diff
@@ -31,25 +31,34 @@ else
...
 echo "Checkout EE 4.0 branch..."
 git branch -D real40 || true
 git checkout -b real40 --track origin/4.0
 
+echo "Creation of image with php 7.3..."
+make php-image-dev
+
+echo "Update composer dependencies"
+make vendor
+
```
Comment (mmetayer) [inline .circleci/detect_structure_changes.sh:55]: this could be removed

File: .circleci/detect_structure_changes.sh [hunk 2]
```diff
@@ -31,25 +31,34 @@ else
...
 echo "Checkout CE 4.0 branch..."
 pushd vendor/akeneo/pim-community-dev
 git branch -D real40 || true
 git checkout -b real40 --track origin/4.0
 popd
```
Comment (mmetayer) [inline .circleci/detect_structure_changes.sh:60]: this could be removed

File: composer.lock [hunk 3]
```diff
@@ -9636,6 +10275,65 @@
...
+            "version": "2.1.0",
+            "source": {
+                "type": "git",
+                "url": "https://github.com/webimpress/safe-writer.git",
+                "reference": "5cfafdec5873c389036f14bf832a5efc9390dcdd"
+            },
+            "dist": {
+                "type": "zip",
+                "url": "https://api.github.com/repos/webimpress/safe-writer/zipball/5cfafdec5873c389036f14bf832a5efc9390dcdd",
+                "reference": "5cfafdec5873c389036f14bf832a5efc9390dcdd",
```
Comment (pierallard) [inline composer.lock:10289]: :shrug:

**Review by pierallard — APPROVED** [review]

**Comment by mmetayer** [comment]
green build EE: https://app.circleci.com/pipelines/github/akeneo/pim-community-dev/15743/workflows/6fab5665-e4c8-45f7-b565-40d085d7c270

**Pull request merged**